pkg/streamingpromql/types/stats.go

```diff
@@ -21,10 +21,63 @@ type QueryStats struct {
 	// For example, if a query is running with a step of 30s with a range vector selector with range 45s,
 	// then samples in the overlapping 15s are counted twice.
 	TotalSamples int64
+
+	TotalSamplesPerStep []int64
+
+	EnablePerStepStats bool
+	timerange          QueryTimeRange
+}
+
+type stepStat struct {
+	T int64
+	V int64
 }
 
 const timestampFieldSize = int64(unsafe.Sizeof(int64(0)))
 
```

Comment:
[nit] I would move this down to be just above `EquivalentFloatSampleCount`, to keep the related bits of code next to each other.